tests/stress/ParameterSweepTest.cpp

```diff
@@ -263,3 +271,101 @@ TEST_P(FCSweepTest, FCTest_Float16) {
+                       glow::ExecutionEngine &EE, size_t numInputs,
+                       size_t numDims, size_t maxLength, size_t axis) {
+  PseudoRNG PRNG;
+  auto &mod = EE.getModule();
+  Function *F = mod.createFunction("main");
+
+  // Make leading dimensions smaller than trailing. Reduces size of tests and is
+  // also in line with typical tests.
+  std::vector<size_t> dims(numDims, maxLength);
+  for (size_t i = 0; i < numDims; i++) {
+    dims[numDims - 1 - i] /= std::pow(2, i);
+  }
+
+  std::vector<NodeValue> inputs(numInputs);
+  for (size_t i = 0; i < numInputs; i++) {
+    auto *IP = mod.createPlaceholder(ElemKind::FloatTy, dims, "input", false);
+    bindings.allocate(IP)->getHandle().randomize(-0.2, 0.2, mod.getPRNG());
+    assert(IP);
+    inputs[i] = IP->getOutput();
+  }
+
+  auto *concat = F->createConcat("concat", inputs, axis);
+  auto *save = F->createSave("save", concat);
+  auto *resultTensor = bindings.allocate(save->getPlaceholder());
+
+  return std::make_pair(F, resultTensor);
+}
+
+/// Helper to test sweeping across a variety of configurations of a Concat by
+/// comparing the results to the Interpreter given some \p allowedError.
+/// \p config contains the backend to compare the Interpreter against, plus the
+/// specific configuration to run for this test. \p interpK and \p backendK are
+/// the element kinds to use for the Interpreter and backend, respectively.
+static void testParamSweepConcat(FourIntTupleConfig config, ElemKind interpK,
+                                 ElemKind backendK, float allowedError) {
+  std::string backend;
+  size_t numInputs, numDims, maxLength, axis;
+  SET_BACKEND_KIND_AND_FOUR_INT_PARAMS(config, backend, numInputs, numDims,
+                                       maxLength, axis);
+  // Exit if axis outside of numDims.
+  if (axis >= numDims) {
+    return;
+  }
+
+  LOG(INFO) << "\n\tTesting Concat with numInputs: " << numInputs
+            << "; numDims: " << numDims << "; maxLength: " << maxLength
+            << "; axis: " << axis << "\n";
+
+  auto boundF =
+      std::bind(createAndInitConcatNet, std::placeholders::_1,
+                std::placeholders::_2, numInputs, numDims, maxLength, axis);
+  compareAgainstInterpreter(backend, boundF, interpK, backendK, allowedError,
+                            parCloneCountOpt);
+}
+
+DECLARE_STATELESS_BACKEND_TEST(ConcatSweepTest, FourIntTupleConfig);
+
+INSTANTIATE_TEST_CASE_P_FOR_BACKEND_COMBINED_TEST(
+    SweepTest, ConcatSweepTest,
+    ::testing::Combine(/* numInputs */ ::testing::Values(1, 2, 4, 8, 16, 32, 64,
+                                                         128, 192, 256),
+                       /* numDims */ ::testing::Range(1, 4),
+                       /* maxLength */ ::testing::Values(16, 32, 64, 128),
+                       /* axis */ ::testing::Range(0, 3)));
+
+/// Compare backend against the interpreter in Float.
+TEST_P(ConcatSweepTest, ConcatTest_Float) {
+  ENABLED_BACKENDS(CPU, OpenCL);
```

Comment:
anything blocking to enable this on Habana as well?

Reply:
#3101